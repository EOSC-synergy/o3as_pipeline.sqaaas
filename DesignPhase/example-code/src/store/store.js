import { configureStore } from '@reduxjs/toolkit'
import plotReducer from './plotSlice'
import modelsReducer from './modelsSlice'
import referenceReducer from './referenceSlice'


export default configureStore({
  reducer: {
    plot: plotReducer,
    models: modelsReducer,
    reference: referenceReducer,
  }
})