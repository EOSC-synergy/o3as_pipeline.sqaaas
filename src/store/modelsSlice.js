import { createSlice } from "@reduxjs/toolkit";

const STATISTICAL_VALUES = ["mean", "median", "derivative", "percentile"];

const initialState = {
    // currently active plot
    plotId: "tco3_zm",
    settings: {
        "tco3_zm": { 
            // this objects holds key-value-pairs, the keys being the model-group 
            // identifier and the values being the settings for each group 
            "all": { 
                // model group storing all information until it is possible 
                // to implement more model groups
                name: "All OCTS models",
                modelList: ["CCMI-1_ACCESS_ACCESS-CCM-refC2"],
                models: { // models is lookup table
                    "CCMI-1_ACCESS_ACCESS-CCM-refC2": { // single model
                        institute: "IMK",
                        dataset: {
                            x: [42],
                            y: [42],
                        },
                        color: "#000000",
                        plotStyle: "solid",
                        isVisible: true, // show/hide individual models from a group
                        mean: true,
                        derivative: true,
                        median: true,
                        percentile: true,
                    }
                },
                hidden: false, // show/hide complete group
                visibileSV: { // lookup table so the reducer impl. can be more convenient
                    mean: true,
                    derivative: true,
                    median: true,
                    percentile: true,
                }
            }
        },
        "tco3_return": {
            "all": { 
                name: "All Return/Recovery models",
                modelList: ["CCMI-1_ACCESS_ACCESS-CCM-refC2"],
                models: { // models is lookup table
                    "CCMI-1_ACCESS_ACCESS-CCM-refC2": { // single model
                        institute: "IMK",
                        dataset: {
                            x: [42],
                            y: [42],
                        },
                        hidden: false, // show/hide individual models from a group
                        mean: true,
                        derivative: true,
                        median: true,
                        percentile: true,
                        color: "#000000",
                        plotStyle: "solid",
                    }
                },
                hidden: false, // show/hide complete group
                visibileSV: { // lookup table so the reducer impl. can be more convenient
                    mean: true,
                    derivative: true,
                    median: true,
                    percentile: true,
                }
            }
        }
    }
}

const modelsSlice = createSlice({
    name: "models",
    initialState,
    reducers: {
        // set models
        // add models

        addModels(state, action) { 
            const { groupId, newModelList }  = action.payload;
            const activeSettings = state.settings[state.plotId];
            const selectedGroup = activeSettings[groupId];
            const activeModelList = selectedGroup.modelList;

            if (selectedGroup === undefined) {
                throw `Tried to access model-group with groupID "${groupId}" that is not yet defined!`;
            }

            // Add model to modelList if it is not already included
            for (var i = 0; i < newModelList.length; i++) {
                if (activeModelList.indexOf(newModelList[i]) !== -1) {
                    continue;
                }

                activeModelList.push(newModelList[i]);
            }

            // ---------------------------------------------------------
            // TODO: Add Model information to the model lookup table
            // ---------------------------------------------------------


        }, // expects groupID, list of models 
        removeModels(state, action) { 
            const { groupId, removeModelList }  = action.payload;
            const activeSettings = state.settings[state.plotId];
            const selectedGroup = activeSettings[groupId];
            const activeModelList = selectedGroup.modelList;

            if (selectedGroup === undefined) {
                throw `Tried to access model-group with groupID "${groupId}" that is not yet defined!`;
            }

            for (var i = 0; i < removeModelList.length; i++) {
                // Remove model from modelList
                activeModelList = activeModelList.filter(item => item !== removeModelList[i]);

                // Delete model object from models
                delete selectedGroup.models[removeModelList[i]];
            }

        }, // expects groupdID, list of models
        updatedModelGroup(state, action) { }, // implementing this requires more knowledge about the UI
        addedModelGroup(state, action) { },
        setVisibility(state, action) { 
            const { groupID, modelID, isVisible } = action.payload;
            const activeSettings = state.settings[state.plotId];
            const activeModel = activeSettings[groupID].models[modelID];
            if (activeModel === undefined) {
                throw `tried to access model with modelID "${modelID}" that is not yet defined!`;
            } 
            activeModel.isVisible = isVisible;
        },
        setStatisticalValuesIncluded(state, action) { // this is for an individual model
            // svType has to be a string of the form: 
            // mean | derivative | median | percentile
            const { groupID, modelID, svType, isIncluded } = action.payload;
            const activeSettings = state.settings[state.plotId];
            const activeModel = activeSettings[groupID].models[modelID];
            if (activeModel === undefined) {
                throw `tried to access model with modelID "${modelID}" that is not yet defined!`;
            }
            if (STATISTICAL_VALUES.includes(svType)) { // svType doesn't represent a valid statistical value
                throw `tried to set statistial value "${svType}" that is not a valid statistical value (${STATISTICAL_VALUES.join("|")})`;
            }
            activeModel[svType] = isIncluded;
        },
        setStatisticalValueForGroup(state, action) { // this is for an entire group
            const { groupID, svType, isIncluded } = action.payload;
            const activeSettings = state.settings[state.plotId];
            if (STATISTICAL_VALUES.includes(svType)) { // svType doesn't represent a valid statistical value
                throw `tried to set statistial value "${svType}" that is not a valid statistical value (${STATISTICAL_VALUES.join("|")})`;
            }
            activeSettings[groupID].visibileSV[svType] = isIncluded;
        },
    }
})

export const { 
    addModels,
    removeModels,
    updatedModelGroup,
    addedModelGroup,
    setVisibility, 
    setStatisticalValuesIncluded, 
    setStatisticalValueForGroup
} = modelsSlice.actions;

export default modelsSlice.reducer;

export const selectCurrentModelGroups = state => state.models[state.plot.plotId];
export const selectCurrentModelGroup = (state, groupId) => state.models[state.plot.plotId][groupId];