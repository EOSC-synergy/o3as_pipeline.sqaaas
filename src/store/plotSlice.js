import { createSlice } from "@reduxjs/toolkit"

/**
 * The initial state of the plotSlice defines the data structure in the 
 * store. Each plot has its own settings i.e. title, location etc.
 */
const initialState = {
    
    plotId: "tco3_zm", // currently active plot
    // maps plotids to their settings
    settings: {
        "tco3_zm": {
            name: "OCTS", // should show up in the drop down menu
            title: "OCTS Plot", // the title shown in the apexcharts generated chart
            location: {
                minLat: -90,
                maxLat: 90
            },
            displayXRange: {
                minX: 1960,
                maxX: 2100,
            },
            displayYRange: {
                minY: 200,
                maxY: 400,
            },
            months: [
                1
            ],
        },
        "tco3_return": {
            name: "Return/Recovery",
            title: "Return/Recovery Plot",
            location: { // custom user defined region
                minLat: -90,
                maxLat: 90
            },
            displayXRange: {
                minX: 0,
                maxX: 0,
            },
            displayYRange: {
                minY: 200,
                maxY: 400,
            },
            months: [
                1
            ],
        }
    }
}

/**
 * The plotSlice is generated by the redux toolkit. The reducers are defined here
 * and the corresponding actions are auto-generated.
 */
const plotSlice = createSlice({
    name: "plot",
    initialState,
    reducers: {
        setActivePlotId(state, action) { // e.g. dispatch(setActivePlotId({id: "tco3_zm"}))
            const { id } = action.payload
            state.plotId = id
        }, 
        setTitle(state, action) { // e.g. dispatch(setActivePlotId({title: "OCTS Plot"}))
            const { title } = action.payload
            state.settings[state.plotId].title = title
        },
        setLocation(state, action) { 
            const {minLat, maxLat} = action.payload
            const location = state.settings[state.plotId].location
            location.minLat = minLat
            location.maxLat = maxLat
        },
        setDisplayXRange(state, action) { 
            const {minX, maxX} = action.payload
            const displayXRange = state.settings[state.plotId].displayXRange
            displayXRange.minX = minX
            displayXRange.maxX = maxX
        },
        setDisplayYRange(state, action) { 
            const {minY, maxY} = action.payload
            const displayYRange = state.settings[state.plotId].displayYRange
            displayYRange.minY = minY
            displayYRange.maxY = maxY
        },
        setMonths(state, action) { 
            state.settings[state.plotId].months = action.payload.months
        },
    }   
})


/**
 * The here listed actions are exported and serve as an interface for
 * the view (our react components).
 */
export const {
    setActivePlotId,
    setTitle,
    setLocation,
    setDisplayXRange,
    setDisplayYRange,
    setMonths,
} = plotSlice.actions

/**
 * The reducer combining all reducers defined in the plot slice. 
 * This has to be included in the redux store, otherwise dispatching 
 * the above defined actions wouldn't trigger state updates.
 */
export default plotSlice.reducer

/**
 * This selectors allows components to select the current plot id
 * from the store. The plot id is a string using the same naming as the
 * o3as api e.g. tco3_zm or tco3_return
 * 
 * @param {object} state the global redux state
 * @returns {string} the current active plot id
 */
export const selectPlotId = state => state.plot.plotId

/**
 * This selectors allows components to select the current plot name
 * from the store. The plot name is the humand readable version specifying 
 * the plot e.g. OCTS or Return/Recovery
 * 
 * @param {object} state the global redux state
 * @returns {string} the current active plot name
 */
export const selectPlotName = state => state.plot.settings[state.plot.plotId].name

/**
 * This selectors allows components to select the current plot title
 * from the store.
 * 
 * @param {object} state the global redux state
 * @returns {string} the current active plot title
 */
export const selectPlotTitle = state => state.plot.settings[state.plot.plotId].title

/**
 * This selectors allows components to select the current plot location
 * from the store. The location is an object containg a minLat and maxLat attribute.
 * 
 * @param {object} state the global redux state
 * @returns {object} holds the current location that includes a minLat and maxLat attribute.  
 */
export const selectPlotLocation = state => state.plot.settings[state.plot.plotId].location

/**
 * This selectors allows components to select the current x range
 * from the store.
 * 
 * @param {object} state the global redux state
 * @returns {object} holds the current x range that includes minX and maxX
 */
export const selectPlotXRange = state => state.plot.settings[state.plot.plotId].displayXRange

/**
 * This selectors allows components to select the current y range
 * from the store.
 * 
 * @param {object} state the global redux state
 * @returns {object} holds the current x range that includes minY and maxY
 */

export const selectPlotYRange = state => state.plot.settings[state.plot.plotId].displayYRange

/**
 * This selectors allows components to select the current selected months
 * from the store.
 * 
 * @param {object} state the global redux state
 * @returns {array} array of strings describing the current selected months
 */

export const selectPlotMonths = state => state.plot.settings[state.plot.plotId].months
