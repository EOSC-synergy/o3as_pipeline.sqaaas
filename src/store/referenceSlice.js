import { createSlice } from "@reduxjs/toolkit"

const initialState = {
    "OCTS": {
        year: 1980,
        model: "CCMI-1_ACCESS_ACCESS-CCM-refC2",
        visibile: false,
        offsetApplied: false,
    }
}

const referenceSlice = createSlice({
    name: "reference",
    initialState,
    reducers: {
        setReferenceYear(state, action) {
            const {year} = action.payload;
            state.OCTS.year = year;
         },
        setReferenceModel(state, action) { 
            const {model} = action.payload;
            state.OCTS.model = model;
        },
        setVisibile(state, action) { },
        setOffsetApplied(state, action) { },
    }
})

export const {
    setReferenceYear,
    setReferenceModel,
    setVisibile,
    setOffsetApplied,
} = referenceSlice.actions

export default referenceSlice.reducer

export const selectCurrentReferenceSettings = state=> state.reference[state.plot.plotId]
