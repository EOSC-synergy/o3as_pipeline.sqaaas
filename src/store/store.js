import { configureStore } from '@reduxjs/toolkit'
import plotReducer from './plotSlice/plotSlice'
import modelsReducer from './modelsSlice/modelsSlice'
import referenceReducer from './referenceSlice'


export default configureStore({
  reducer: {
    plot: plotReducer,
    models: modelsReducer,
    reference: referenceReducer,
  }
})