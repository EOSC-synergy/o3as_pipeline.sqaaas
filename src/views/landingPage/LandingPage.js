import * as React from 'react';
import Graph from './Graph/Graph';
import Sidebar from './Sidebar/Sidebar';
import Button from '@mui/material/Button';
import PropTypes from "prop-types";
import { FormControl, Grid } from '@mui/material';
import { height, width } from '@mui/system';
import EditModelGroupModal from './Sidebar/InputComponents/ModelGroupConfigurator/EditModelGroupModal/EditModelGroupModal';

/**
 * main view of web page
 * @param {Object} props
 * @param {function} props.reportError - function to report an error
 * @returns {JSX} a jsx containing all main components
 */
function LandingPage(props) {

    const [isSidebarOpen, setSidebarOpen] = React.useState(false);
    
    /**
     * Function to open sidebar
     */
    const openSidebar = () => {
        setSidebarOpen(true);
    }

    /**
     * Function to close sidebar,
     * if the user does not currently try to navigate the sidebar
     * @param {event} event the event that triggered the call of this function
     */
    const closeSidebar = (event) => {
        // for accessibility do not close sidebar if users 
        // try to navigate sidebar using Tab or Shift
        if (
            event &&
            event.type === 'keydown' &&
            (event.key === 'Tab' || event.key === 'Shift')
          ) {
            return;
          }
        setSidebarOpen(false);
    }
    
    return (
    <div data-testid="landingPage" style={{width: "100%"}}> 
      <Sidebar reportError={props.reportError} isOpen={isSidebarOpen} onClose={closeSidebar} onOpen={openSidebar} />
      <Button variant="outlined" onClick={openSidebar} style={{width: "100%"}}>Open Sidebar (Dev)</Button>
  
      <div style={{display: "flex", justifyContent: "center", alignItems: "center"}}>
        <div style={{width: "95%"}}>
          <Graph reportError={props.reportError} />
        </div>
      </div>

      <Button variant="outlined" onClick={openSidebar} style={{width: "100%"}}/>
      <EditModelGroupModal isOpen={true} onClose={()=>{}}/>
      
    </div>
    );
}

LandingPage.propTypes = {
    reportError: PropTypes.func,
}

export default LandingPage;