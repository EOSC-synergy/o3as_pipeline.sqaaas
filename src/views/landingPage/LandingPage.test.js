import React from 'react';
import ReactDOM from 'react-dom';
import LandingPage from './LandingPage';
import {render, screen} from '@testing-library/react';
import '@testing-library/jest-dom';
import userEvent from '@testing-library/user-event';
import { queryByTestId } from '@testing-library/dom';
import { Provider } from "react-redux";
import { createTestStore } from '../../store/store';

jest.mock('react-apexcharts', () => {
  return {
    __esModule: true,
    default: () => {
      return <div />
    },
  }
})

let store;
describe('test LandingPage component rendering', () => {
  beforeEach(() => {
    store = createTestStore();
  });

  it('renders without crashing', () => {
    render(
      <Provider store={store}>
        <LandingPage reportError={() => {}} isSidebarOpen={true} openSidebar={jest.fn()} closeSidebar={jest.fn()} />
      </Provider>
    );
  });
  
  // Snapshot test
  it('LandingPage renders correctly', () => {
      const { container } = render(
        <Provider store={store}>
            <LandingPage reportError={() => {}} isSidebarOpen={true} openSidebar={jest.fn()} closeSidebar={jest.fn()} />
        </Provider>
      );
      expect(container).toMatchSnapshot();
  });

  it('closes sidebar, when clicking outside of sidebar', () => {
    const closeSidebar = jest.fn();
    const { getByTestId, queryByTestId } = render(
      <Provider store={store}>
          <LandingPage reportError={() => {}} isSidebarOpen={true} openSidebar={jest.fn()} closeSidebar={closeSidebar} />
      </Provider>
    );
    userEvent.click(getByTestId(/landingPage-not-sidebar/));
    expect(closeSidebar).toHaveBeenCalled();
  });
});