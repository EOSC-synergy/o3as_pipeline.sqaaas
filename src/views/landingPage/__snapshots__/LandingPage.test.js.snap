// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LandingPage renders correctly 1`] = `
Array [
  <button
    className="MuiButton-root MuiButton-outlined MuiButton-outlinedPrimary MuiButton-sizeMedium MuiButton-outlinedSizeMedium MuiButtonBase-root  css-1rwt2y5-MuiButtonBase-root-MuiButton-root"
    disabled={false}
    onBlur={[Function]}
    onClick={[Function]}
    onContextMenu={[Function]}
    onDragLeave={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    onKeyUp={[Function]}
    onMouseDown={[Function]}
    onMouseLeave={[Function]}
    onMouseUp={[Function]}
    onTouchEnd={[Function]}
    onTouchMove={[Function]}
    onTouchStart={[Function]}
    tabIndex={0}
    type="button"
  >
    Open Sidebar (Dev)
  </button>,
  <div
    className="MuiDrawer-root MuiDrawer-docked css-19iwfmc-MuiDrawer-docked"
    data-testid="sidebar"
  >
    <div
      className="MuiPaper-root MuiPaper-elevation MuiPaper-elevation0 MuiDrawer-paper MuiDrawer-paperAnchorRight MuiDrawer-paperAnchorDockedRight css-18sg6k4-MuiPaper-root-MuiDrawer-paper"
      style={
        Object {
          "pointerEvents": "",
          "visibility": "hidden",
        }
      }
    >
      <div
        className="css-cfiwj7"
      >
        <button
          className="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeMedium css-78trlr-MuiButtonBase-root-MuiIconButton-root"
          data-testid="sidebarClose"
          disabled={false}
          onBlur={[Function]}
          onClick={[Function]}
          onContextMenu={[Function]}
          onDragLeave={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          tabIndex={0}
          type="button"
        >
          <svg
            aria-hidden={true}
            className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
            data-testid="CloseIcon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M19 6.41 17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12z"
            />
          </svg>
        </button>
      </div>
      PlotTypeSelector
      <div
        className="MuiPaper-root MuiPaper-elevation MuiPaper-elevation0 MuiAccordion-root css-1dm7ii1-MuiPaper-root-MuiAccordion-root"
        data-testid="section"
      >
        <div
          aria-controls="panel1a-content"
          aria-expanded={false}
          className="MuiButtonBase-root MuiAccordionSummary-root css-1icm36a-MuiButtonBase-root-MuiAccordionSummary-root"
          id="panel1a-header"
          onBlur={[Function]}
          onClick={[Function]}
          onContextMenu={[Function]}
          onDragLeave={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          role="button"
          tabIndex={0}
        >
          <div
            className="MuiAccordionSummary-content css-1betqn-MuiAccordionSummary-content"
          >
            <p
              className="MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root"
            >
              APPEARANCE
            </p>
          </div>
          <div
            className="MuiAccordionSummary-expandIconWrapper css-yw020d-MuiAccordionSummary-expandIconWrapper"
          >
            <svg
              aria-hidden={true}
              className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
              data-testid="ExpandMoreIcon"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M16.59 8.59 12 13.17 7.41 8.59 6 10l6 6 6-6z"
              />
            </svg>
          </div>
        </div>
        <div
          className="MuiCollapse-root MuiCollapse-vertical MuiCollapse-hidden css-bz4dnt-MuiCollapse-root"
          style={
            Object {
              "minHeight": "0px",
            }
          }
        >
          <div
            className="MuiCollapse-wrapper MuiCollapse-vertical css-smkl36-MuiCollapse-wrapper"
          >
            <div
              className="MuiCollapse-wrapperInner MuiCollapse-vertical css-9l5vo-MuiCollapse-wrapperInner"
            >
              <div
                aria-labelledby="panel1a-header"
                className="MuiAccordion-region"
                id="panel1a-content"
                role="region"
              >
                <div
                  className="MuiAccordionDetails-root css-ta84vm-MuiAccordionDetails-root"
                >
                  PlotNameField
                  XAxisSlider
                  YAxisSlider
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
      <div
        className="MuiPaper-root MuiPaper-elevation MuiPaper-elevation0 MuiAccordion-root css-1dm7ii1-MuiPaper-root-MuiAccordion-root"
        data-testid="section"
      >
        <div
          aria-controls="panel1a-content"
          aria-expanded={false}
          className="MuiButtonBase-root MuiAccordionSummary-root css-1icm36a-MuiButtonBase-root-MuiAccordionSummary-root"
          id="panel1a-header"
          onBlur={[Function]}
          onClick={[Function]}
          onContextMenu={[Function]}
          onDragLeave={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          role="button"
          tabIndex={0}
        >
          <div
            className="MuiAccordionSummary-content css-1betqn-MuiAccordionSummary-content"
          >
            <p
              className="MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root"
            >
              MODELS
            </p>
          </div>
          <div
            className="MuiAccordionSummary-expandIconWrapper css-yw020d-MuiAccordionSummary-expandIconWrapper"
          >
            <svg
              aria-hidden={true}
              className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
              data-testid="ExpandMoreIcon"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M16.59 8.59 12 13.17 7.41 8.59 6 10l6 6 6-6z"
              />
            </svg>
          </div>
        </div>
        <div
          className="MuiCollapse-root MuiCollapse-vertical MuiCollapse-hidden css-bz4dnt-MuiCollapse-root"
          style={
            Object {
              "minHeight": "0px",
            }
          }
        >
          <div
            className="MuiCollapse-wrapper MuiCollapse-vertical css-smkl36-MuiCollapse-wrapper"
          >
            <div
              className="MuiCollapse-wrapperInner MuiCollapse-vertical css-9l5vo-MuiCollapse-wrapperInner"
            >
              <div
                aria-labelledby="panel1a-header"
                className="MuiAccordion-region"
                id="panel1a-content"
                role="region"
              >
                <div
                  className="MuiAccordionDetails-root css-ta84vm-MuiAccordionDetails-root"
                >
                  ModelGroupConfigurator
                  AddModelGroupModal
                  EditModelGroupModal
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
      <div
        className="MuiPaper-root MuiPaper-elevation MuiPaper-elevation0 MuiAccordion-root css-1dm7ii1-MuiPaper-root-MuiAccordion-root"
        data-testid="section"
      >
        <div
          aria-controls="panel1a-content"
          aria-expanded={false}
          className="MuiButtonBase-root MuiAccordionSummary-root css-1icm36a-MuiButtonBase-root-MuiAccordionSummary-root"
          id="panel1a-header"
          onBlur={[Function]}
          onClick={[Function]}
          onContextMenu={[Function]}
          onDragLeave={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          role="button"
          tabIndex={0}
        >
          <div
            className="MuiAccordionSummary-content css-1betqn-MuiAccordionSummary-content"
          >
            <p
              className="MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root"
            >
              FILTER DATA
            </p>
          </div>
          <div
            className="MuiAccordionSummary-expandIconWrapper css-yw020d-MuiAccordionSummary-expandIconWrapper"
          >
            <svg
              aria-hidden={true}
              className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
              data-testid="ExpandMoreIcon"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M16.59 8.59 12 13.17 7.41 8.59 6 10l6 6 6-6z"
              />
            </svg>
          </div>
        </div>
        <div
          className="MuiCollapse-root MuiCollapse-vertical MuiCollapse-hidden css-bz4dnt-MuiCollapse-root"
          style={
            Object {
              "minHeight": "0px",
            }
          }
        >
          <div
            className="MuiCollapse-wrapper MuiCollapse-vertical css-smkl36-MuiCollapse-wrapper"
          >
            <div
              className="MuiCollapse-wrapperInner MuiCollapse-vertical css-9l5vo-MuiCollapse-wrapperInner"
            >
              <div
                aria-labelledby="panel1a-header"
                className="MuiAccordion-region"
                id="panel1a-content"
                role="region"
              >
                <div
                  className="MuiAccordionDetails-root css-ta84vm-MuiAccordionDetails-root"
                >
                  <div
                    className="MuiDivider-root MuiDivider-fullWidth MuiDivider-withChildren css-1ws8119-MuiDivider-root"
                    role="separator"
                  >
                    <span
                      className="MuiDivider-wrapper css-qywfm8-MuiDivider-wrapper"
                    >
                      <p
                        className="MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root"
                      >
                        LATITUDE BAND
                      </p>
                    </span>
                  </div>
                  <div
                    className="MuiBox-root css-1n3sbe4"
                  >
                    <div
                      className="MuiFormControl-root css-q8hpuo-MuiFormControl-root"
                    >
                      <div
                        className="MuiOutlinedInput-root MuiInputBase-root MuiInputBase-colorPrimary MuiInputBase-formControl css-9ddj71-MuiInputBase-root-MuiOutlinedInput-root"
                        onClick={[Function]}
                      >
                        <div
                          aria-expanded="false"
                          aria-haspopup="listbox"
                          aria-labelledby="latitudeBandSelectorLabel latitudeBandSelector"
                          className="MuiSelect-select MuiSelect-outlined MuiOutlinedInput-input MuiInputBase-input css-11u53oe-MuiSelect-select-MuiInputBase-input-MuiOutlinedInput-input"
                          id="latitudeBandSelector"
                          onBlur={[Function]}
                          onFocus={[Function]}
                          onKeyDown={[Function]}
                          onMouseDown={[Function]}
                          role="button"
                          tabIndex={0}
                        >
                          <span
                            className="notranslate"
                            dangerouslySetInnerHTML={
                              Object {
                                "__html": "&#8203;",
                              }
                            }
                          />
                        </div>
                        <input
                          aria-hidden={true}
                          className="MuiSelect-nativeInput css-yf8vq0-MuiSelect-nativeInput"
                          disabled={false}
                          onAnimationStart={[Function]}
                          onChange={[Function]}
                          required={false}
                          tabIndex={-1}
                          value="0,0"
                        />
                        <svg
                          aria-hidden={true}
                          className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium MuiSelect-icon MuiSelect-iconOutlined css-hfutr2-MuiSvgIcon-root-MuiSelect-icon"
                          data-testid="ArrowDropDownIcon"
                          focusable="false"
                          viewBox="0 0 24 24"
                        >
                          <path
                            d="M7 10l5 5 5-5z"
                          />
                        </svg>
                        <fieldset
                          aria-hidden={true}
                          className="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
                        >
                          <legend
                            className="css-1z7n62"
                          >
                            <span>
                              LatitudeBand
                            </span>
                          </legend>
                        </fieldset>
                      </div>
                      <label
                        className="MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-outlined MuiFormLabel-root MuiFormLabel-colorPrimary MuiFormLabel-filled css-1kty9di-MuiFormLabel-root-MuiInputLabel-root"
                        data-shrink={true}
                        id="latitudeBandSelectorLabel"
                      >
                        Latitude Band
                      </label>
                    </div>
                  </div>
                  TimeCheckBoxGroup
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
      <div
        className="MuiPaper-root MuiPaper-elevation MuiPaper-elevation0 MuiAccordion-root css-1dm7ii1-MuiPaper-root-MuiAccordion-root"
        data-testid="section"
      >
        <div
          aria-controls="panel1a-content"
          aria-expanded={false}
          className="MuiButtonBase-root MuiAccordionSummary-root css-1icm36a-MuiButtonBase-root-MuiAccordionSummary-root"
          id="panel1a-header"
          onBlur={[Function]}
          onClick={[Function]}
          onContextMenu={[Function]}
          onDragLeave={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          role="button"
          tabIndex={0}
        >
          <div
            className="MuiAccordionSummary-content css-1betqn-MuiAccordionSummary-content"
          >
            <p
              className="MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root"
            >
              REFERENCE YEAR
            </p>
          </div>
          <div
            className="MuiAccordionSummary-expandIconWrapper css-yw020d-MuiAccordionSummary-expandIconWrapper"
          >
            <svg
              aria-hidden={true}
              className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
              data-testid="ExpandMoreIcon"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M16.59 8.59 12 13.17 7.41 8.59 6 10l6 6 6-6z"
              />
            </svg>
          </div>
        </div>
        <div
          className="MuiCollapse-root MuiCollapse-vertical MuiCollapse-hidden css-bz4dnt-MuiCollapse-root"
          style={
            Object {
              "minHeight": "0px",
            }
          }
        >
          <div
            className="MuiCollapse-wrapper MuiCollapse-vertical css-smkl36-MuiCollapse-wrapper"
          >
            <div
              className="MuiCollapse-wrapperInner MuiCollapse-vertical css-9l5vo-MuiCollapse-wrapperInner"
            >
              <div
                aria-labelledby="panel1a-header"
                className="MuiAccordion-region"
                id="panel1a-content"
                role="region"
              >
                <div
                  className="MuiAccordionDetails-root css-ta84vm-MuiAccordionDetails-root"
                >
                  ReferenceModelSelector
                  ReferenceYearSlider
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
      <button
        className="MuiButton-root MuiButton-outlined MuiButton-outlinedPrimary MuiButton-sizeMedium MuiButton-outlinedSizeMedium MuiButtonBase-root  css-126gm7o-MuiButtonBase-root-MuiButton-root"
        disabled={false}
        onBlur={[Function]}
        onClick={[Function]}
        onContextMenu={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        tabIndex={0}
        type="button"
      >
        Download
      </button>
    </div>
  </div>,
  <div>
    Graph
  </div>,
]
`;
